daemon/api_sideload_n_try_test.go

```diff
@@ -645,6 +645,128 @@ func (s *sideloadSuite) TestCleanUpUnusedTempSnapFiles(c *check.C) {
 	c.Check(matches, check.HasLen, 1)
 }
 
+func (s *sideloadSuite) TestSideloadManySnaps(c *check.C) {
+	d := s.daemonWithFakeSnapManager(c)
+	expectedFlags := &snapstate.Flags{RemoveSnapPath: true, DevMode: true}
+
+	restore := daemon.MockSnapstateInstallPathMany(func(_ context.Context, s *state.State, infos []*snap.SideInfo, paths []string, userID int, flags *snapstate.Flags) ([]*state.TaskSet, error) {
+		c.Check(flags, check.DeepEquals, expectedFlags)
+		c.Check(userID, check.Not(check.Equals), 0)
+
+		var tss []*state.TaskSet
+		for i, path := range paths {
+			si := infos[i]
+			c.Check(path, testutil.FileEquals, si.RealName)
+
+			ts := state.NewTaskSet(s.NewTask("fake-install-snap", fmt.Sprintf("Doing a fake install of %q", si.RealName)))
+			tss = append(tss, ts)
+		}
+
+		return tss, nil
+	})
+	defer restore()
+
+	snaps := []string{"one", "two"}
+	var i int
+	readRest := daemon.MockUnsafeReadSnapInfo(func(string) (*snap.Info, error) {
+		info := &snap.Info{SuggestedName: snaps[i]}
+		i++
+		return info, nil
+	})
+	defer readRest()
+
+	body := "----hello--\r\n" +
+		"Content-Disposition: form-data; name=\"devmode\"\r\n" +
+		"\r\n" +
+		"true\r\n" +
+		"----hello--\r\n"
```

Comment:
It'd be nice to have a test with dangerous too.

We have this pattern that may you'll find useful:

```
func (s *sideloadSuite) testSideloadManySnaps(c *check.C, flagsBody string, expectedFlags snapstate.Flags) {

}

func (s *sideloadSuite) TestSideloadManySnapsDevmode(c *check.C) {
	body := "----hello--\r\n" +
		"Content-Disposition: form-data; name=\"devmode\"\r\n" +
		"\r\n" +
		"true\r\n" +
		"----hello--\r\n"
	expectedFlags := snapstate.Flags{RemoveSnapPath: true, DevMode: true}
	s.testSideloadManySnaps(c,  *check.C, body, expectedFlags)
}

func (s *sideloadSuite) TestSideloadManySnapsDangerous(c *check.C) {
	body := "----hello--\r\n" +
		"Content-Disposition: form-data; name=\"dangerous\"\r\n" +
		"\r\n" +
		"true\r\n" +
		"----hello--\r\n"
	expectedFlags := snapstate.Flags{RemoveSnapPath: true}
	s.testSideloadManySnaps(c,  *check.C, body, expectedFlags)
}

func (s *sideloadSuite) TestSideloadManySnapsAsserted(c *check.C) {
	// TODO setup assertions
	body := "----hello--\r\n"
	expectedFlags := snapstate.Flags{RemoveSnapPath: true}
	s.testSideloadManySnaps(c,  *check.C, body, expectedFlags)
}
```

Reply:
yes, would be good